### PR TITLE
Refactor summary row

### LINKED ISSUE #187: Refactor Headers in Summary Table
Users should be able to toggle, "Stats", "Values" and "Meta" on and off as desired (instead of just picking one at a time) 

Request from @agstn 

# Test Notes

* confirm that missing data rates are shown when missing > 0%. Click ADTIMELINES_partialMissing.csv on [this page](https://rawgit.com/RhoInc/web-codebook/refactor-summary-row/build/test-page/explorer.html)
* Confirm that missing rates <10% are gray, >10% are red (0% rates are hidden)
* Confirm that controls for "header details" have been removed
* Confirm that a text-based summary of the variable appears above the chart when a row is expanded.
  * categorical variables should have 'missing' and 'unique' on the left and a small data table on the right
  * continuous variables should have 'missing', 'mean', 'SD' and 'median' on the left and a list of extreme values on the right. 
* Confirm that, when provided, metadata tags are shown below the chart. Click a file with metadata included = yes on [this page](https://rawgit.com/RhoInc/web-codebook/refactor-summary-row/build/test-page/explorer.html)

### PR DESCRIPTION
**Pull request by jwildfire**
This PR covers most of the new user-facing functionality in v1.4.0. Test notes are in the issues listed below. 

closes #187 #251 #252 #256 #258 #259 #260

### CODE REVIEW

File: src/charts/spark/makeHist.js [hunk 1]
```diff
@@ -0,0 +1,82 @@
+export default function makeHist(this_, d) {
```
Comment (samussiah) [inline src/charts/spark/makeHist.js]: Super slick.

File: src/codebook/summaryTable/renderRow/makeTitle.js [hunk 2]
```diff
@@ -14,20 +18,50 @@ export default function makeTitle(d) {
...
+    .datum(d)
+    .each(createSpark);
+
+  //add percent missing (if > 0%)
+  d3select(this)
+    .append('span')
+    .attr('class', 'percent-missing')
+    .text(d => d3format('0.1%')(d.statistics.percentMissing) + ' missing')
+    .style('display', d => (d.statistics.percentMissing == 0 ? 'none' : null))
+    .style('cursor', 'pointer')
```
Comment (samussiah) [inline src/codebook/summaryTable/renderRow/makeTitle.js]: To filter at a click event?

File: src/codebook/title/updateCountSummary.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
+import { format as d3format } from 'd3';
```
Comment (samussiah) [inline src/codebook/title/updateCountSummary.js]: Makes a lot more sense up here.

**Comment by danedexF5** [comment]
#187: 
-"Confirm that a text-based summary of the variable appears above the chart when a row is expanded" Not seeing this when a variable is expanded.
- Test page isn't rendering in IE

**Comment by jwildfire** [comment]
@danedexF5 Can you please make an issue to cover the IE bug? I'll take a look ASAP and decide if we need to deal with it in v1.4 or if it can wait.

**Comment by danedexF5** [comment]
@jwildfire Sure, I'll create an issue now. To clarify a little, I restarted and I can get one test page to come up in IE (https://rawgit.com/RhoInc/web-codebook/refactor-summary-row/build/test-page/) and one still will not (https://rawgit.com/RhoInc/web-codebook/refactor-summary-row/build/test-page/explorer.html). So it appears the issue is with the explorer part of it.

**Comment by jwildfire** [comment]
#187 just refers to this section of the variable summary (text-based might not be the best identifier. screen shot 2018-05-31 at 11 53 13 am

**Comment by jwildfire** [comment]
#251 yep, I think the test you describe above confirms functionality for the column counts.

**Comment by jwildfire** [comment]
#252 @danedexF5 Can you please create a bug and document the steps to reproduce the situation when a missing value >10% isn't red.

**Comment by jwildfire** [comment]
#256 I think the color difference in the header charts is probably too subtle to notice. Going to change the categorical bars to a dark blue and see if that makes them easier to distinguish.

**Comment by danedexF5** [comment]
#252 @jwildfire disregard my note on this one. I was mistaken and not converting to %'s, so it appears to be working as expected. 10% and above is red, below 10% is grey.

**Pull request merged**